### DISCUSSION TITLE
About template studio.

### DISCUSSION THREAD

**Discussion by Petrarca181** [microsoft/TemplateStudio#4663] [General]
I recently installed template studio, and I was a bit shocked. I'm new in programming and for me code looks a bit overcomplicated.  Like shell page and page services, what I see [here](https://learn.microsoft.com/en-us/windows/apps/design/controls/navigationview)  and how it's realized in  template studio are two different worlds.  If  in my app I have 5-6 page do I actually need that setup and all those host services?  Or they are needed to support 100% mvvm  in the app? Thank you.

**Comment by mikebattista** [comment]
You can try removing what you don't want but most of it is needed.

**Comment by rmotta01** [comment]
Overengineered.
